src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1128,7 +1128,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       def adaptApplyInsertion(): Tree = doAdaptApplyInsertion(retry = false)
 
       def doAdaptApplyInsertion(retry: Boolean): Tree =
-        if (currentRun.isScala3Cross && !isPastTyper && tree.symbol != null && tree.symbol.isModule && tree.symbol.companion.isCase && isFunctionType(pt))
+        if (currentRun.isScala3X && !isPastTyper && tree.symbol != null && tree.symbol.isModule && tree.symbol.companion.isCase && isFunctionType(pt))
```

Comment:
I realize that apply insertion should be enabled by default, not even under `-Xsource:3`.

For a library built with `-Xsoruce:3-cross`, case class companions don't extend `FunctionN`. This should not break `List(1).map(A)` for users of the library that use the default compiler options.

@som-snytt wdyt?